app/models/question.rb

```diff
@@ -0,0 +1,61 @@
+class Question < ApplicationRecord
+  enum question_type: [:drafted, :published]
+  ###ASSOCIATION###
+  belongs_to :author, class_name: 'BaseUser', foreign_key: :base_user_id
+
+  has_many_attached :pdfs
+
+  has_many :questions_topics, dependent: :destroy
+  has_many :topics, through: :questions_topics
+  has_many :related_users, -> { distinct }, through: :topics, source: 'users'
+  has_many :notifications, dependent: :destroy
+
+  ###CALLBACKS###
+   before_save :check_title_uniqueness
+   before_save :add_url_slug
+   after_create :set_question_notifications
+   after_create :send_notifications
```

Comment:
What if a question is created in draft state and later on published. In that scenario, notifications will not be broadcasted.